mergefs.go

```diff
@@ -25,3 +28,64 @@ func (mfs MergedFS) Open(name string) (fs.File, error) {
 	}
 	return nil, os.ErrNotExist
 }
+
+// ReadDir reads from the directory, and produces a DirEntry array of different
+// directories.
+//
+// It iterates through all different filesystems that exist in the mfs MergeFS
+// filesystem slice and it identifies overlapping directories that exist in different
+// filesystems
+func (mfs MergedFS) ReadDir(name string) ([]fs.DirEntry, error) {
+	dirsMap := make(map[string]fs.DirEntry)
+	for _, filesystem := range mfs.filesystems {
+		if fsys, ok := filesystem.(fs.ReadDirFS); ok {
```

Comment:
This whole function could be made much simpler by just calling:

`fs.ReadDir(filesystem, name)`

https://golang.org/pkg/io/fs/#ReadDir

It encapsulates all the fiddly stuff for you

Reply:
Not entirely sure either - we have to intercept the normal `fs.ReadDir` to include:

```
for _, v := range dir {
				if _, ok := dirsMap[v.Name()]; !ok {
					dirsMap[v.Name()] = v
				}
			}
```

since what this PR is for, no?

Reply:
Yeah tried that many times and then just copy-pasted the standard implementation with some custom bits in it. Changed now.

Reply:
awesome, thanks. it looks much cleaner now